src/Transport.ts

```diff
@@ -61,6 +54,13 @@ export class Transport {
    */
   private world: string = '';
 
+  /**
+   * Status connection behavior subject.
+   * Internally keeps track of the connection state.
+   * Uses a Behavior Subject because it has an initial state and stores a value.
+   */
+  private status$ = new BehaviorSubject<string>('disconnected');
+
```

Comment:
Small nit; you can reduce the diff here by putting this back where it was.

Reply:
Standard linter rules complain when the order of fields is mixed. All `private` fields should be below the `public` ones and so forth. While we are not using a linter here, it's better to keep those styles in mind.